Tests/Common/Scheduling/DateRulesTests.cs

```diff
@@ -243,12 +396,85 @@ public void EndOfWeekWithSymbol()
             {
                 count++;
                 Assert.IsTrue(date.DayOfWeek == DayOfWeek.Friday || date.DayOfWeek == DayOfWeek.Thursday);
-                Console.WriteLine(date + " " + date.DayOfWeek);
             }
 
             Assert.AreEqual(52, count);
         }
 
+        [Test]
+        public void EndOfWeekWithSymbolWithOffset()
+        {
+            var rules = GetDateRules();
+            var rule = rules.WeekEnd(Symbols.SPY, 2);
+            var dates = rule.GetDates(new DateTime(2000, 01, 01), new DateTime(2000, 12, 31));
+
+            int count = 0;
+            foreach (var date in dates)
+            {
+                count++;
+
+                // Friday - 2 = Wednesday
+                Assert.IsTrue(date.DayOfWeek == DayOfWeek.Wednesday);
+            }
+
+            Assert.AreEqual(52, count);
+        }
+
+        [TestCase(5)] // Friday - 5 = Sunday
+        [TestCase(6)] // Friday - 6 = Saturday
+        public void EndOfWeekWithSymbolWithOffsetToWeekend(int offset)
+        {
+            var rules = GetDateRules();
+            var rule = rules.WeekEnd(Symbols.SPY, offset);
+            var dates = rule.GetDates(new DateTime(2000, 01, 01), new DateTime(2000, 3, 31));
+
+            int count = 0;
+            foreach (var date in dates)
+            {
+                count++;
+
+                // Both test cases are non-tradable days for Spy
+                // We expect it to find a previous day in the week that is tradable, meaning
+                // Friday in this set of dates
+                Assert.IsTrue(date.DayOfWeek == DayOfWeek.Friday);
+            }
+
+            Assert.AreEqual(13, count);
+        }
+
+        [Test]
+        public void DateRulesExpectedNames()
+        {
+            var rules = GetDateRules();
+            IDateRule rule;
+
+            // WeekEnd Rules
+            rule = rules.WeekEnd();
+            Assert.AreEqual("WeekEnd", rule.Name);
+
+            rule = rules.WeekEnd(1);
+            Assert.AreEqual("WeekEnd-1", rule.Name);
+
+            rule = rules.WeekEnd(Symbols.SPY);
+            Assert.AreEqual("SPY: WeekEnd", rule.Name);
+
+            rule = rules.WeekEnd(Symbols.SPY, 2);
+            Assert.AreEqual("SPY: WeekEnd-2", rule.Name);
+
+            // WeekStart rules
+            rule = rules.WeekStart();
+            Assert.AreEqual("WeekStart", rule.Name);
+
+            rule = rules.WeekStart(1);
+            Assert.AreEqual("WeekStart+1", rule.Name);
+
+            rule = rules.WeekStart(Symbols.SPY);
+            Assert.AreEqual("SPY: WeekStart", rule.Name);
+
+            rule = rules.WeekStart(Symbols.SPY, 3);
+            Assert.AreEqual("SPY: WeekStart+3", rule.Name);
+        }
+
```

Comment:
Nice tests! 👍 
I think it would be nice to add some Crypto tests for which the market never closes